### PR TITLE
feat: Add support for specifying conrol plane subnets separate from those used by node groups (data plane)

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for specifying conrol plane subnets separate from those used by node groups (data plane)
- Update version `1.21` references to `1.22`
- Update K8s auth API version from `v1alpha1` to `v1beta1` in examples 

## Motivation and Context
- Setting small, specific subnets for the EKS control plane allows users to expand their data plane later on by adding additional subnets to the data plane. However, you cannot modify the EKS control plane subnets as this will cause the control plane to be replaced/recreated which is disruptive. This is useful for large clusters (aka - "jumbo clusters") where VPCs might be stacking CIDR ranges and additional subnets added to the data plane to support the required number of IPs

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -331,6 +332,7 @@ module "vpc" {
   azs             = ["${local.region}a", "${local.region}b", "${local.region}c"]
   private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
   public_subnets  = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+  intra_subnets   = ["10.0.7.0/28", "10.0.7.16/28", "10.0.7.32/28"]
```
Comment (antonbabenko) [inline examples/complete/main.tf:335]: A good use-case for `intra_subnets`!

File: variables.tf [hunk 2]
```diff
@@ -44,8 +44,14 @@ variable "cluster_additional_security_group_ids" {
   default     = []
 }
 
+variable "control_plane_subnet_ids" {
+  description = "A list of subnet IDs where the EKS cluster control plane (ENIs) will be provisioned. Used for expanding the pool of subnets used by nodes/node groups without replacing the EKS control plane"
```
Comment (antonbabenko) [inline variables.tf:48]: `ENIs` = Elastic Network Interfaces, or what?

**Review by antonbabenko — APPROVED** [review]
LGTM

File: variables.tf (hunk 2 above)
Comment (bryantbiggs) [reply variables.tf:48]: Correct - Elastic Network Interfaces

**Comment by antonbabenko** [comment]
This PR is included in version 18.24.0 :tada:

**Comment by MRostanski** [comment]
Kudos @bryantbiggs! That is good scalability opportunity.

**Pull request merged**